src/test/ui/rfc-2632-const-trait-impl/tilde-const-invalid-places.rs

```diff
@@ -17,6 +17,12 @@ fn rpit_assoc_bound() -> impl IntoIterator<Item: ~const T> { Some(S) }
 fn apit_assoc_bound(_: impl IntoIterator<Item: ~const T>) {}
 //~^ ERROR `~const` is not allowed
 
+fn generic<P: ~const T>() {}
+//~^ ERROR `~const` is not allowed
+
+fn where_clause<P>() where P: ~const T {}
+//~^ ERROR `~const` is not allowed
```

Comment:
Afaict from the impl, these functions would be accepted if they were in an impl block. They should only be allowed if the impl block is for a `const Trait` impl.